### ISSUE TITLE
Servo should support using OpenXR on Linux via Monado

### ISSUE THREAD

**Issue by RangerMauve** [servo/webxr#192]
There's a growing community of OpenXR projects building on Linux around [monado](https://monado.freedesktop.org/).

It would be nice if Servo supported this so that people could easily experiment with WebXR on Linux based systems compared to the closed source Windows/Mac/Oculus environments.

**Comment by jdm** [comment]
I agree this would be a good improvement. It will require some way of indicating the desired backend (D3D or GL) when creating the OpenXrDiscovery struct, and restructuring some of the code like [this](https://github.com/servo/webxr/blob/cc6af6d8f4244dd5523a2dbc1f49d5f60b054f57/webxr/openxr/mod.rs#L479-L486) to support non-windows platforms.

**Comment by RangerMauve** [comment]
For what it's worth, https://github.com/Ralith/openxrs supports Monado and it looks like the author of that lib is actively in the Monado Discord.

**Comment by technobaboo** [comment]
Currently working on this right now, any advice?

**Comment by jdm** [comment]
What kinda ND of advice are you looking for?

**Comment by technobaboo** [comment]
really just any ideas, tips, tricks... you know this codebase better than I
do, I just know Rust and OpenXR and OpenXRs...

**Comment by jdm** [comment]
I think it generally requires the following:
* make the openxr backend of the webxr crate support linux (see all uses of d3d in https://github.com/servo/webxr/blob/master/webxr/openxr/mod.rs)
* integrate the openxr backend into linux builds in https://github.com/servo/servo/blob/aa427dd4122359901b5d7b613bc54f64fdb94364/ports/libsimpleservo/api/src/lib.rs#L827-L897

**Comment by technobaboo** [comment]
Okay just as I thought then, thanks so much for confirming that those are
the 2 spots to focus on!

**Comment by svillar** [comment]
Also if you need some examples of OpenXR usage on real browsers for inspiration you can check [Chromium's implementation](https://source.chromium.org/chromium/chromium/src/+/main:device/vr/openxr/) and also [Wolvic's implementation](https://github.com/Igalia/wolvic/tree/main/app/src/openxr/cpp)
